### PR TITLE
refactor(cardinal): cleanup of profiling. (part 1)

### PR DESCRIPTION
**Pull request by pyrofolium**
Closes: WORLD-791

Profiling was a bit messy as it was designed at first to serve some specific profiling requests as well. There were several systems that people wanted tested and it was designed so each person could specify what configuration of profile they wanted to run. (This was for us, world-engine developers)

The code has now been made to run a single profiling run and overall configuration and quit when finished. The system used for this involves a the creation of a tree based component. We can evolve the nature of what is to be profiled later but for now this is just code organization so that we can get automated profiling to run via the CI. 

My eventual goal is to get an svg that looks like this: 
https://www.brendangregg.com/FlameGraphs/cpu-bash-flamegraph.svg

into every PR thread. Also just to be clear I don't want profiling to be a requirement to finish before merging. I just want some bot to post the above svg into the PR thread.

### CODE REVIEW

**Comment by pyrofolium** [comment]
cpu-bash-flamegraph
Testing...

**Comment by pyrofolium** [comment]
> cpu-bash-flamegraph Testing...

This test ended up not working. The svg has interactive elements and exposing it as raw assets as part of the repo disables javascript.

**Comment by pyrofolium** [comment]
https://argus-labs.github.io/world-engine-profiles/other.html

File: e2e/testgames/gamebenchmark/sys/sys.go
Comment (technicallyty) [inline e2e/testgames/gamebenchmark/sys/sys.go:1]: can you add some comments to some of these fns? `ProfilerTerminatorDecoratorForSystem` might be confusing at first glance for newcomers

**Review by technicallyty — APPROVED** [review]
LGTM

**Pull request closed without merging**